test/Vanguard.test.ts

```diff
@@ -0,0 +1,212 @@
+import { ethers, ignition } from 'hardhat'
+import {
+  GovernorRootstockCollective,
+  RIFToken,
+  StRIFToken,
+  VotingVanguardsRootstockCollective,
+} from '../typechain-types'
+import VotingVanguardsModule from '../ignition/modules/VotingVanguardsModule'
+import { deployContracts } from './deployContracts'
+import { expect } from 'chai'
+import { SignerWithAddress } from '@nomicfoundation/hardhat-ethers/signers'
+import { randomBytes, hexlify, formatEther } from 'ethers'
+import { loadFixture, mine } from '@nomicfoundation/hardhat-network-helpers'
+import { VoteType } from '../types'
+
+const ZERO_BYTE = '0x00'
+const rootstockGasPriceAverage = 63228564n
+
+describe('Vanguard NFT', () => {
+  // The number of proposals that need to be checked to determine whether the user voted for any of them
+  const proposalCount = 3
+  const ipfsFolderCid = 'QmZYHgFMjZ9SNvFwP9rCxtDEgF2JfJwQtaSg2fqxTX31eG'
+  const stRifThreshold = 50n * 10n ** 18n
+  const votingPower = 100n * 10n ** 18n
+  const maxSupply = 1000
+  const mintLimit = 200
+
+  let deployer: SignerWithAddress
+  let voter: SignerWithAddress
+  let proposalTarget: SignerWithAddress
+
+  before(async () => {
+    ;[deployer, voter, proposalTarget] = await ethers.getSigners()
+  })
+
+  // give voting power
+  const enfranchise = async (rif: RIFToken, stRIF: StRIFToken) => {
+    // give voting power to deployer to be able to create a proposal
+    await rif.approve(await stRIF.getAddress(), votingPower).then(tx => tx.wait())
+    await stRIF.depositAndDelegate(deployer.address, votingPower).then(tx => tx.wait())
+    // give voting power to voter
+    await rif.transfer(voter.address, votingPower).then(tx => tx.wait())
+    await rif
+      .connect(voter)
+      .approve(await stRIF.getAddress(), votingPower)
+      .then(tx => tx.wait())
+    await stRIF
+      .connect(voter)
+      .depositAndDelegate(voter.address, votingPower)
+      .then(tx => tx.wait())
+  }
+
+  const deploy = async () => {
+    const contracts = await deployContracts()
+    const vanguard = (
+      await ignition.deploy(VotingVanguardsModule, {
+        parameters: {
+          VanguardNFT: {
+            maxSupply,
+            mintLimit,
+            stRifThreshold,
+            stRif: await contracts.stRIF.getAddress(),
+            governor: await contracts.governor.getAddress(),
+            proposalCount,
+            ipfsFolderCid,
+          },
+        },
+      })
+    ).VanguardNFT as unknown as VotingVanguardsRootstockCollective
+    await enfranchise(contracts.rif, contracts.stRIF)
+    return { ...contracts, vanguard }
+  }
+
+  const createProposal = async (governor: GovernorRootstockCollective) => {
```

Comment:
why did you not test on `TESTNET`? Hardhat network is not guaranteed to have the same amount of gas